File: tutorials/testing/java/reddit_snippet.java
import java.util.ArrayList;
import bridges.connect.Bridges;
import bridges.connect.DataSource;
import bridges.data_src_dependent.Reddit;


// This program fragment illustrates how to access the reddit data
public class reddit_snippet {
	public static void main(String[] args) throws Exception {

		// create Bridges object
		Bridges bridges = new Bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
    
		DataSource ds = bridges.getDataSource();

		ArrayList<String> subreddits = ds.getAvailableSubreddits();

		// get some data from the  "Ask Science" subreddit
		ArrayList<Reddit> reddit_posts = ds.getRedditData("askscience");
		int count = 0;
		for (Reddit r: reddit_posts) {
		    ++count;
		    if (count > 10)
			break;
		    System.out.println("ID: "+r.getID());
		    System.out.println("Title: "+r.getTitle());
		    System.out.println("Author: "+r.getAuthor());
		    System.out.println("Score: "+r.getScore());
		    System.out.println("VoteRatio: "+r.getVoteRatio());
		    System.out.println("CommentCount: "+r.getCommentCount());
		    System.out.println("Subreddit: "+r.getSubreddit());
		    System.out.println("PostTime: "+r.getPostTime());
		    System.out.println("URL: "+r.getURL());
		    System.out.println("Text: "+r.getText());
		    System.out.println("");
		}
		
	}
}

File: tutorials/testing/c++/reddit_snippet.cpp

#include <iostream>
#include <string>
#include <vector>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/Reddit.h"

using namespace bridges;

// This program fragment illustrates how to access and read the Reddit Post data 
int main(int argc, char **argv) {

	// create Bridges object
	Bridges bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// create data source object
	DataSource ds (&bridges);

	vector<Reddit> reddit_list = ds.getRedditData("askscience");
	std::cout<<""<<reddit_list.size()<<" post returned"<<"\n";
	std::cout<<"Printing at most 10\n";
	int count = 0;
	for (auto post : reddit_list) {
	  ++count;
	  if (count >10)
	    break;
	  std::cout<<"ID: "<<post.getID()<<"\n";
	  std::cout<<"Title: "<<post.getTitle()<<"\n";
	  std::cout<<"Author: "<<post.getAuthor()<<"\n";
	  std::cout<<"Score: "<<post.getScore()<<"\n";
	  std::cout<<"VoteRatio: "<<post.getVoteRatio()<<"\n";
	  std::cout<<"CommentCount: "<<post.getCommentCount()<<"\n";
	  std::cout<<"Subreddit: "<<post.getSubreddit()<<"\n";
	  std::cout<<"PostTime: "<<post.getPostTime()<<"\n";
	  std::cout<<"URL: "<<post.getURL()<<"\n";
	  std::cout<<"Text: "<<post.getText()<<"\n";
	  std::cout<<"\n";
	}
	
	
	return 0;
}

